frontend/src/components/LeadModal/LeadModal.js

```diff
@@ -56,7 +60,144 @@ export const LeadModal = ({ open, onClose, addLead }) => {
   const [instagram, setInstagram] = useState('');
   const [linkedin, setLinkedin] = useState('');
   const [twitter, setTwitter] = useState('');
-  const [requiredField, setRequiredField] = useState(false);
+  const [users, setUsers] = useState([]);
+
+  const checkForErrors = (response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else {
+      throw new Error('Something went wrong');
+    }
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      try {
+        const token = localStorage.getItem('partnerFinderToken');
+        const headers = {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        };
+
+        const response = await fetch(`${API_HOST}/users`, {
+          headers: headers,
+        });
+        const data = await checkForErrors(response);
+        setUsers(data.users);
+      } catch (err) {
+        console.log(err.message);
+      }
+    };
+    getUsers();
+  }, []);
+
+  const checkValidPhone = (number) => {
+    const extractedNums = number.replace(/[^0-9]/g, '');
+    if (extractedNums.length !== 10) {
+      return false;
+    } else {
+      return extractedNums;
+    }
+  };
+
+  const checkValidEmail = (address) => {
+    const validEmail = new RegExp('^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$');
+    return validEmail.test(address);
+  };
+
+  const checkValidUrl = (url) => {
+    var validUrl = new RegExp(
+      '^(https?:\\/\\/)?' + // protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '(\\#[-a-z\\d_]*)?$',
+      'i'
+    );
+    return validUrl.test(url);
+  };
```

Comment:
Wonder if such utils exist in a NPM package we could rely on? If not, would we ever want to re-use these utilities in another file? Might a better place for them be in a utility file?